pkg/yang/find.go

```diff
@@ -21,33 +21,28 @@ import (
 	"strings"
 )
 
-// trimPrefix trims the current module's prefix from the given name. If the
+// localPrefix returns the local prefix used by the containing (sub)module to
+// refer to its own module.
+func localPrefix(n Node) string {
+	return RootNode(n).GetPrefix()
+}
+
+// trimLocalPrefix trims the current module's prefix from the given name. If the
 // name is not prefixed with it, the same string is returned unchanged.
-// TODO(borman): we need to properly handle prefixs and not depend on
-// not having collisions.
-func trimPrefix(n Node, name string) string {
-	parts := strings.Split(name, ":")
-	if len(parts) == 1 {
-		return name
-	}
-	// We should have a single *Module parent.  It contains our prefix.
-	for {
-		if m, ok := n.(*Module); ok {
-			if m.Prefix != nil && m.Prefix.Name == parts[0] {
-				return parts[1]
-			}
-			return name
-		}
-		n = n.ParentNode()
+func trimLocalPrefix(n Node, name string) string {
+	pfx := localPrefix(n)
```

Comment:
Can `localPrefix` ever return nil? It seems like it'd be an error to have no prefix defined? (I'm OK that we don't handle it, I'm just trying to work out when the below condition is required.

Reply:
Oh, I see - you're expecting that `name` includes something like "foo:bar", and this is to trim `foo:`. 

Could you update the docstring to say that if the name is not prefixed with the local module, or is unprefixed entirely, then the same thing is returned?